modules/ossm-exposing-service-using-istio-gateway-and-virtualservice.adoc

```diff
@@ -0,0 +1,156 @@
+oc apply -n httpbin -f https://raw.githubusercontent.com/istio/istio/refs/heads/release-1.24/samples/httpbin/httpbin.yaml
+----
+
+. Create a YAML file named `httpbin-gw.yml` that defines an Istio `Gateway` resource. This resource configures gateway proxies to expose port 80 (HTTP) for the host, `httpbin.example.com`.
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1
+kind: Gateway
+metadata:
+  name: httpbin-gateway
+  namespace: httpbin
+spec:
+  selector:
+    istio: <gateway_name> <1>
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - httpbin.example.com <2>
+----
+<1> Set the `selector` to the unique label or set of labels specified in the pod template of the gateway proxy `Deployment`. By default, the {istio} `Gateway` resource configuration will apply to matching gateway pods in all namespaces.
+<2> Using the `hosts` field, specify a list of addresses that can be used by clients when attempting to access a mesh service at the associated port.
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f httpbin-gw.yml
+----
+
+. Create a YAML file named `httpbin-vs.yml` for a `VirtualService` The `VirtualService` defines the rules that route traffic from the gateway proxy to the `httpbin` service.
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1
+kind: VirtualService
+metadata:
+  name: httpbin
+  namespace: httpbin
+spec:
+  hosts:
+  - httpbin.example.com <1>
+  gateways:
+  - httpbin-gateway <2>
+  http:
+  - match:
+    - uri:
+        prefix: /status
+    - uri:
+        prefix: /headers
+    route:
+    - destination: <3>
+        port:
+          number: 8000
+        host: httpbin
+----
+<1> Specify the `hosts` that the routing rules of the `VirtualService` will be applied to. The `hosts` specified must be exposed by the {istio} `Gateway` resource the VirtualService is bound to.
+<2> Bind the `VirtualService` to the {istio} `Gateway` resource created in the previous step by adding the `Gateway` name to the list of gateways.
+<3> Route matching traffic to the `httpbin` service deployed earlier by defining a `destination` that includes the `host` and `port` of the httpbin `Service`.
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f httpbin-vs.yml
+----
+
+. For verification purposes, deploy a `curl` client in a namespace outside of the mesh.
++
+[source,terminal]
+----
+oc create namespace curl
+oc apply -n curl -f https://raw.githubusercontent.com/istio/istio/refs/heads/release-1.24/samples/curl/curl.yaml
+----
+
+. Using the `curl` client, send a request to the `/headers` endpoint of the httpbin application through the ingress gateway `Service`. Set the `Host` header of the request to `httpbin.example.com` to match the host that the {istio} `Gateway` and `VirtualService` resources specify.
```

Comment:
```suggestion
. Using the `curl` client, send a request to the `/headers` endpoint of the `httpbin` application through the ingress gateway `Service`. Set the `Host` header of the request to `httpbin.example.com` to match the host that the {istio} `Gateway` and `VirtualService` resources specify.
```

I see it varies in the core OCP documentation.

Reply:
done